crucible.mjs

```diff
@@ -70,16 +70,72 @@ Hooks.once("init", async function() {
   /**
    * Configurable properties of the system which affect its behavior.
    */
+  function itemPacksByType(type) {
+      let potentialPacks = {};
+      for (const pack of game.packs) {
+        if (pack.metadata.type !== "Item") continue;
+        for (const item of pack.index) {
+          if (item.type === type) {
+            let name = pack.metadata.label;
+            let folder = pack.folder;
+            while (folder) {
+              name = `${folder.name} -> ${name}`
+              folder = folder.folder;
+            }
+            potentialPacks[pack.metadata.id] = name;
+            break;
+          }
+        }
+      }
+      return potentialPacks;
+  }
+  game.settings.register("crucible", "ancestrySources", {
+    name: "SETTINGS.AncestrySourcesName",
+    hint: "SETTINGS.AncestrySourcesHint",
+    scope: "world",
+    config: true,
+    type: new foundry.data.fields.SetField(new foundry.data.fields.StringField({ required: true, choices: () => itemPacksByType("ancestry")})),
+    default: [SYSTEM.COMPENDIUM_PACKS.ancestry],
+    requiresReload: true,
+  });
+  game.settings.register("crucible", "backgroundSources", {
+    name: "SETTINGS.BackgroundSourcesName",
+    hint: "SETTINGS.BackgroundSourcesHint",
+    scope: "world",
+    config: true,
+    type: new foundry.data.fields.SetField(new foundry.data.fields.StringField({ required: true, choices: () => itemPacksByType("background")})),
+    default: [SYSTEM.COMPENDIUM_PACKS.background],
+    requiresReload: true,
+  });
+  game.settings.register("crucible", "spellSources", {
+    name: "SETTINGS.SpellSourcesName",
+    hint: "SETTINGS.SpellSourcesHint",
+    scope: "world",
+    config: true,
+    type: new foundry.data.fields.SetField(new foundry.data.fields.StringField({ required: true, choices: () => itemPacksByType("spell")})),
+    default: [SYSTEM.COMPENDIUM_PACKS.spell],
+    requiresReload: true,
+  });
+  game.settings.register("crucible", "talentSources", {
+    name: "SETTINGS.TalentSourcesName",
+    hint: "SETTINGS.TalentSourcesHint",
+    scope: "world",
+    config: true,
+    type: new foundry.data.fields.SetField(new foundry.data.fields.StringField({ required: true, choices: () => itemPacksByType("talent")})),
+    default: [SYSTEM.COMPENDIUM_PACKS.talent],
+    requiresReload: true,
+  });
+  
   crucible.CONFIG = {
     /**
      * Configuration of compendium packs which are used as sources for system workflows.
      * @type {Record<string, Set<string>>}
      */
     packs: {
-      ancestry: new Set([SYSTEM.COMPENDIUM_PACKS.ancestry]),
-      background: new Set([SYSTEM.COMPENDIUM_PACKS.background]),
-      spell: new Set([SYSTEM.COMPENDIUM_PACKS.spell]),
-      talent: new Set([SYSTEM.COMPENDIUM_PACKS.talent]),
+      ancestry: game.settings.get("crucible", "ancestrySources"),
```

Comment:
I think we need some fallback logic that if this setting `Set` is empty, it falls back to the default value. For example if `crucible.ancestrySources` is an empty set, that needs to mean "use the default ancestries" rather than "there are no ancestries".